### PR TITLE
[CARBONDATA-3856] Support the LIMIT operator for show segments command

### PR DESCRIPTION
**Pull request by marchpure**
### Why is this PR needed?
 When the number segments is large, "SHOW SEGMENTS" time overhead is too high. The LIMIT operator shall be supported in the SHOW SEGMENTS command.
 
 ### What changes were proposed in this PR?
Add LIMIT operator in the grammar parsing and read segments function.
    
 ### Does this PR introduce any user interface change?
Yes

 ### Is any new testcase added?
Yes

### CODE REVIEW

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1443/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3168/

File: integration/spark/src/test/scala/org/apache/carbondata/spark/testsuite/segment/ShowSegmentTestCase.scala [hunk 1]
```diff
@@ -191,7 +214,11 @@ class ShowSegmentTestCase extends QueryTest with BeforeAndAfterAll {
     sql("drop table if exists a")
     sql("create table a(a string) stored as carbondata")
     sql("insert into a select 'k'")
+    sql("insert into a select 'j'")
+    sql("insert into a select 'k'")
     val rows = sql("show segments for table a").collect()
+    assert(sql(s"show segments for table a").collect().length == 3)
```
Comment (Indhumathi27) [inline integration/spark/src/test/scala/org/apache/carbondata/spark/testsuite/segment/ShowSegmentTestCase.scala:220]:
```suggestion
    assert(rows.length == 3)
```

File: integration/spark/src/test/scala/org/apache/carbondata/spark/testsuite/segmentreading/TestSegmentReading.scala [hunk 2]
```diff
@@ -251,13 +251,16 @@ class TestSegmentReading extends QueryTest with BeforeAndAfterAll {
         s"""LOAD DATA local inpath '$resourcesPath/data.csv' INTO TABLE carbon_table_show_seg OPTIONS
             |('DELIMITER'= ',', 'QUOTECHAR'= '\"')""".stripMargin)
       val df = sql("SHOW SEGMENTS for table carbon_table_show_seg as select * from carbon_table_show_seg_segments")
+      sql("SHOW SEGMENTS for table carbon_table_show_seg as select * from carbon_table_show_seg_segments").show()
```
Comment (Indhumathi27) [inline integration/spark/src/test/scala/org/apache/carbondata/spark/testsuite/segmentreading/TestSegmentReading.scala:254]: Can remove this line

File: docs/segment-management-on-carbondata.md [hunk 3]
```diff
@@ -54,6 +54,12 @@ concept which helps to maintain consistency of data and easy transaction managem
   SHOW SEGMENTS ON CarbonDatabase.CarbonTable
   ```
 
+  Show lastest 10 visible segments
```
Comment (Indhumathi27) [inline docs/segment-management-on-carbondata.md:57]: After compaction also, it should show latest segments right? For example, there are 6 segments, after major compaction, show segments with limit 2 will display 4.1 and 5th segment only, whereas latest segments are 0.2 and 0.3. I think we should get latest segments based on timestamp.

File: integration/spark/src/test/scala/org/apache/carbondata/spark/testsuite/segment/ShowSegmentTestCase.scala (hunk 1 above)
Comment (marchpure) [reply integration/spark/src/test/scala/org/apache/carbondata/spark/testsuite/segment/ShowSegmentTestCase.scala]: modified

File: integration/spark/src/test/scala/org/apache/carbondata/spark/testsuite/segmentreading/TestSegmentReading.scala (hunk 2 above)
Comment (marchpure) [reply integration/spark/src/test/scala/org/apache/carbondata/spark/testsuite/segmentreading/TestSegmentReading.scala]: modified

File: docs/segment-management-on-carbondata.md (hunk 3 above)
Comment (marchpure) [reply docs/segment-management-on-carbondata.md]: Yeah. currently, the showed segments are ordered by loadname desc. in the type of：
+---+---------+-----------------------+
|ID |Status   |Load Start Time        |
+---+---------+-----------------------+
|5  |Compacted|2020-06-18 04:20:09.041|
|4.1|Success      |2020-06-18 04:20:09.041|
|4  |Compacted|2020-06-18 04:20:08.69 |
|3  |Compacted|2020-06-18 04:20:07.622|
|2.1|Compacted|2020-06-18 04:20:07.622|
|2  |Compacted|2020-06-18 04:20:07.226|

Comment (Indhumathi27) [reply docs/segment-management-on-carbondata.md]: yes. i think it is better to display latest segments based on timestamp on limit

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3169/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1444/

**Comment by Indhumathi27** [comment]
LGTM

**Pull request closed without merging**